app/scripts/controllers/permissions/background-api.test.js

```diff
@@ -359,15 +477,24 @@ describe('permission background API methods', () => {
         revokePermission: jest.fn(),
       };
 
+      const accountsController = {
+        getAccountByAddress: jest.fn().mockReturnValue({
+          address: '0xdeadbeef',
+          scopes: ['eip155:0'],
+        }),
+      };
+
       getPermissionBackgroundApiMethods({
         permissionController,
+        accountsController,
       }).removePermittedAccount('foo.com', '0xdeadbeef');
 
       expect(permissionController.updateCaveat).not.toHaveBeenCalled();
       expect(permissionController.revokePermission).not.toHaveBeenCalled();
     });
 
-    it('revokes the entire permission if the removed account is the only eip:155 scoped account', () => {
+    // TODO: review intended behavior here
+    it('revokes the entire permission if the removed account is the only account', () => {
```

Comment:
awkwardness here with `wallet` and `wallet:eip155`.

Reply:
This may not matter because the UI force you to explicitly disconnect if all accounts are unselected